### ISSUE TITLE
How to classify Images

### ISSUE THREAD

**Issue by Mrinal18** [wangyirui/AD_Prediction#8]
Hey @wangyirui ,can you tell me how i can classify a test image? i have trained the model but can't figure out how to classify the images with accurate answers

**Comment by wangyirui** [comment]
Sorry, I don't fully understand your question. If you mean how to output the predict label, all you need is:
```
train_output = model(imgs)
_, predict = train_output.topk(1)
```
Since the output of your model is the logprobability of belonging to each class. So we use 'argmax' to obtain the highest probability's index, the label.

**Comment by Mrinal18** [comment]
@wangyirui that is helpful and thank you for such a fast reply.Although, I would like to know if i input any arbiturary image, how will it classify the image. and what are the steps for that?

**Comment by wangyirui** [comment]
@Mrinal18 What means " what are the steps for that"? The computation path or how to write code to do that?

**Comment by wangyirui** [comment]
@Mrinal18 Actually, we don't have code to do that. But it's not difficult, I can give you some brief step to do that, and you may want to implement it by yourself.
1. Load the model as you did previously.
2. adapt our code to read in the pic (you may need to modify how to read img since currently we read in img according to train.txt and test.ext)
3. copy the validation part, remove the loss computation and directly use topk(1) to generate labels.

That's what in my mind now.

**Comment by Mrinal18** [comment]
@wangyirui that is very helpful, although then how did you implement the classification part? since in your report, in page 6 ` MRI scan will be predicted as “having AD”.` is written. how did you predict this?

**Comment by Mrinal18** [comment]
@wangyirui, is it possible if you can share a video where output can be seen? i just need it for reference. Thank you

**Comment by wangyirui** [comment]
[predict.zip](https://github.com/wangyirui/AD_Prediction/files/1981373/predict.zip)

Check this, I modified the code for you (may have some minor bugs, but the overall procedure should like this)

**Comment by Mrinal18** [comment]
wow, this is awesome, i will try to run it. 
Meanwhile i ran the system and printed "vote" and here is the output.

```
('this is vote', Variable containing:
    0     1     0
    0     0     0
    0     0     1
    1     0     0
    0     0     0
    0     0     0
    0     1     0
    0     0     0
    0     0     0
    0     0     0
    0     0     0
    1     0     0
    1     0     0
    0     0     0
    0     1     0
    0     0     0
    0     0     0
    0     0     0
    0     0     0
    0     1     0
    0     1     0
    0     0     0
    0     0     0
    0     0     0
    0     1     1
    0     0     0
    0     0     0
    0     0     0
    0     0     0
    0     0     0
    0     1     0
    0     0     0
[torch.LongTensor of size 32x3]
)
```

Is this output correct? and what does it signify?

**Comment by Mrinal18** [comment]
And this is the final vote 

```
('this is the final vote', Variable containing:
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
 0
[torch.LongTensor of size 32]
)

```


PS: i ran main_alextnet.py

**Comment by wangyirui** [comment]
@Mrinal18 I don't remember the details of it. But logic is: we use different slices to make predictions, then we vote the result to reduce the prediction variance. So, based on this idea, I hope you can figure out what the vote should be.  I glanced at the first result, it may not be correct. I will be very busy from today on, so I may not reply you as quick as before.

**Comment by Mrinal18** [comment]
@wangyirui , i understand you are busy. thank you for your help and such quick replies. i will post few doubts if i am not able to understand something. kindly reply when you get time

**Comment by Mrinal18** [comment]
@wangyirui, is it okay if you can send me a demo on this project of yours? i think that will help me a lot as i might be missing something small that you guys have already figured out. it is a request since this project is super important. it would be great if you can help me with it.

**Comment by Mrinal18** [comment]
Also, @wangyirui can you kindly uploard `/autoencoder_pretrained_model19` pretrained model, which is used in `main_autoencoder.py`
